omp/btx_ompinterval_callbacks.cpp

```diff
@@ -0,0 +1,139 @@
+#include "magic_enum.hpp"
+#include "ompt.h.include"
+#include "xprof_utils.hpp"
+#include <iostream>
+#include <metababel/metababel.h>
+#include <string>
+#include <unordered_map>
+
+using hpt_function_name_omp_t =
+    std::tuple<hostname_t, process_id_t, thread_id_t, std::string>;
+
+struct data_s {
+  std::unordered_map<hpt_function_name_omp_t, uint64_t> host_start;
+};
+typedef struct data_s data_t;
+
+static void btx_initialize_component(void **usr_data) {
+  *usr_data = new data_t;
+}
+
+static void btx_finalize_component(void *usr_data) {
+  delete static_cast<data_t *>(usr_data);
+}
+
+// returns start_ts, for use in traffic or other messages on scope end
+static int64_t _host_op_callback(void *btx_handle, void *usr_data, int64_t ts,
```

Comment:
also nitpicking but no need for `_`  :) it's static so not exported and nothing is generated so cannot have some name conflict

Reply:
fair point, renamed.